### PR TITLE
chore: improve yeoman-ui extension activation time

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: packages/backend/src/utils/env.ts [hunk 1]
```diff
@@ -27,24 +27,36 @@ export class GeneratorNotFoundError extends Error {
...
+  }
+
+  private getEnvNpmPath(): Promise<string[]> {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        // this operation takes up to 2 seconds
+        // it should be wrapped with setTimeout to provide promise like behaviour
+        resolve(this.createEnvInstance().getNpmPaths());
+      }, 5);
+    });
```
Comment (alex-gilin) [inline packages/backend/src/utils/env.ts:44]: please explain. why 5 milisec ? it seems tricky ...

File: packages/backend/src/utils/npm.ts [hunk 2]
```diff
@@ -51,8 +48,14 @@ class Command {
...
-  private async execCommand(arg: string): Promise<any> {
-    return promisify(exec)(arg);
+  private async execCommand(arg: string): Promise<string> {
+    const result: ExecResult = await promisify(exec)(arg);
+    if (!_.isEmpty(result.stderr)) {
+      // no need to throw error, because stderr usually contains warnings
+      getConsoleWarnLogger().warn(result.stderr);
+    }
+
+    return result.stdout;
```
Comment (alex-gilin) [inline packages/backend/src/utils/npm.ts:58]: in Window exec can added some chars ('\n' or etc.)  to its stdout result

File: packages/backend/src/utils/npm.ts [hunk 3]
```diff
@@ -135,8 +142,15 @@ class Command {
     }
   }
 
-  public getGlobalNodeModulesPath(): string {
-    return this.globalNodeModulesPath;
+  private async getGlobalPath(): Promise<string> {
+    const globalNodeModulesPath = await this.getGlobalNodeModulesPath();
+    const nmLength = path.join(path.sep, "node_modules").length;
+    return globalNodeModulesPath.substring(0, globalNodeModulesPath.length - nmLength);
```
Comment (alex-gilin) [inline packages/backend/src/utils/npm.ts:148]: risky method. better try :
  _.join(_.without(_.split(globalNodeModulesPath, path.sep), 'node_modules'), path.sep) 
up to you

**Review by alex-gilin — CHANGES_REQUESTED** [review]

File: packages/backend/src/utils/env.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/env.ts:44]: Even 1 mills is enough.
I need the setTimeout to work and create promise

File: packages/backend/src/utils/npm.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/npm.ts:58]: that is why when I use result of execCommand I do _.trim()

Comment (slavik-lvovsky) [reply packages/backend/src/utils/npm.ts:58]: Done

File: packages/backend/src/utils/npm.ts (hunk 3 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/npm.ts:148]: Done

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**